### PR TITLE
Additional loggers on debug level

### PR DESCRIPTION
**Pull request by jdziurzynski**
KNTK - 262

Added logging of API requests and responses as suggested in:
https://github.com/kentik/terraform-provider-kentik-synthetics/pull/81#issuecomment-1042346692

Here https://pastebin.com/MB6YQqy0 is a part of debug output containing new loggers.

### CODE REVIEW

**Review by mmac-m3a — APPROVED** [review]
Looks reasonable.

**Comment by mmac-m3a** [comment]
I'll wait for Daniel's approval before merging the code.

File: internal/provider/resource_cloudexport.go
Comment (danielfurman) [inline internal/provider/resource_cloudexport.go:39]:
```suggestion
	tflog.Debug(ctx, "Create cloud export Kentik API request"request", req)
```
I am not convinced about encoding data to JSON, because it might distort information about the object, that might be useful for debugging (because of `omitempty`, custom marshallers, etc.). If `req` is not enough consider using fmt.Sprintf("%+v", req) to display names of struct fields.

Also, failing to prepare data only for logging **should not** result in failure of the provider.

**Review by danielfurman — CHANGES_REQUESTED** [review]

File: internal/provider/data_source_cloudexport_item.go
Comment (danielfurman) [inline internal/provider/data_source_cloudexport_item.go:25]:
```suggestion
	tflog.Debug(ctx, "Get cloud export Kentik API response", "response", getResp)
```
Could you show if the output of the log is correct? I think the Debug() API requires key-value  pairs after the message.

In other places too.

File: internal/provider/data_source_cloudexport_item.go
Comment (danielfurman) [reply internal/provider/data_source_cloudexport_item.go]: Done, but added a separate question

File: internal/provider/data_source_cloudexport_item.go
Comment (danielfurman) [reply internal/provider/data_source_cloudexport_item.go]: Done

File: internal/provider/resource_cloudexport.go
Comment (danielfurman) [reply internal/provider/resource_cloudexport.go]: Done

File: internal/provider/data_source_cloudexport_item.go
Comment (danielfurman) [reply internal/provider/data_source_cloudexport_item.go]: Done

**Review by danielfurman — APPROVED** [review]

**Review by mmac-m3a — APPROVED** [review]

**Pull request merged**
